File: 342_Power_of_Four.py
class Solution:
    def isPowerOfFour(self, n: int) -> bool:
        if n < 1:
            return False
        
        while n % 4 == 0:
            n //= 4
        return n == 1
        

a = Solution()
print(a.isPowerOfFour(16))  # True
print(a.isPowerOfFour(5))   # False
print(a.isPowerOfFour(1))   # True

File: 342_Power_of_Four.c
#include<stdio.h>
#include<stdbool.h>

bool isPowerOfFour(int n) {
    if (n <= 0) {
        return false;
    }
    
    while (n % 4 == 0) {
        n /= 4;
    }

    return n == 1;
}

int main() {
    int n;
    printf("Enter a number: ");
    scanf("%d", &n);
    
    if (isPowerOfFour(n)) {
        printf("%d is a power of four.\n", n);
    } else {
        printf("%d is not a power of four.\n", n);
    }
    
    return 0;
}